components/eamxx/src/share/io/scream_scorpio_interface.F90

```diff
@@ -76,6 +76,8 @@ module scream_scorpio_interface
             eam_update_time,             & ! Update the timestamp (i.e. time variable) for a given pio netCDF file
             get_int_attribute,           & ! Retrieves an integer global attribute from the nc file
             set_int_attribute,           & ! Writes an integer global attribute to the nc file
+            set_str_attribute,           & ! Writes an string global attribute to the nc file
+            set_variable_attribute,      & ! Writes an string attribute for a variable in the nc file
```

Comment:
This seems very similar to `set_variable_metadata`. Maybe we can use that one?

Reply:
Good point.  I don't think `set_variable_metadata` will work if the "definition" phase has been closed.  I think we can probably use `set_variable_metadata` but enhanced to reopen the definition phase if it has been closed, with the `set_X_attribute` functions do.

Quick question, I didn't see `set_variable_metadata` before you mentioned it here because when I started looking for ways to update metadata I first saw `set/get_int_attribute`.  Should we adopt the `set_x_attribute` naming convention for all functions that set metadata?  Or change the name of `set_int_attribute` to `set_int_metadata`?

Reply:
I have redone these a bit more neatly in the branch where I resume filling the output files. Here, you can just get rid of one of the two, to avoid redundancies, and then I'll deal with conflicts when issuing the next PR.

Reply:
sounds good.